grafana/resource_team_external_group.go

```diff
@@ -0,0 +1,165 @@
+	}
+}
+
+func CreateTeamExternalGroup(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	teamID := d.Get("team_id").(int)
+	d.SetId(strconv.FormatInt(int64(teamID), 10))
+	if err := UpdateGroups(d, meta); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return diag.Diagnostics{}
+}
+
+func ReadTeamExternalGroup(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	if err := ReadGroups(d, meta); err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
+}
+
+func UpdateTeamExternalGroup(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	if err := UpdateGroups(d, meta); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return diag.Diagnostics{}
+}
+
+func DeleteTeamExternalGroup(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	if err := UpdateGroups(d, meta); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return diag.Diagnostics{}
+}
+
+func ReadGroups(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*client).gapi
+	teamID, _ := strconv.ParseInt(d.Id(), 10, 64)
+	teamGroups, err := client.TeamGroups(teamID)
+	if err != nil {
+		return err
+	}
+	groupSlice := []string{}
+	for _, teamGroup := range teamGroups {
+		groupSlice = append(groupSlice, teamGroup.GroupID)
+	}
+	d.Set("groups", groupSlice)
+
+	return nil
+}
+
+func UpdateGroups(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*client).gapi
+
+	addGroups, removeGroups := groupChanges(d)
+	teamID, _ := strconv.ParseInt(d.Id(), 10, 64)
+
+	for _, group := range addGroups {
+		err := client.NewTeamGroup(teamID, group)
+		if err != nil {
+			return err
+		}
+	}
+
+	for _, group := range removeGroups {
+		err := client.DeleteTeamGroup(teamID, group)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func groupChanges(d *schema.ResourceData) ([]string, []string) {
+	// Get the lists of team groups read in from Grafana state (old) and configured (new)
+	state, config := d.GetChange("groups")
+
+	stateGroups := make([]string, state.(*schema.Set).Len())
+	for _, v := range state.(*schema.Set).List() {
+		stateGroups = append(stateGroups, v.(string))
+	}
+
+	configGroups := make([]string, config.(*schema.Set).Len())
+	for _, v := range config.(*schema.Set).List() {
+		configGroups = append(configGroups, v.(string))
+	}
+
+	addGroups := []string{}
+	for _, group := range configGroups {
+		_, exists := SliceFind(stateGroups, group)
+		if !exists {
+			addGroups = append(addGroups, group)
+		}
+	}
+
+	removeGroups := []string{}
+	for _, group := range stateGroups {
+		_, exists := SliceFind(configGroups, group)
+		if !exists {
+			removeGroups = append(removeGroups, group)
+		}
+	}
+
+	return addGroups, removeGroups
+}
+
+// SliceFind takes a slice and looks for an element in it. If found it will
+// return it's key, otherwise it will return -1 and a bool of false.
+func SliceFind(slice []string, val string) (int, bool) {
```

Comment:
You don't seem to ever be using the returned `int`, consider making this simple returning a `bool` instead.